springboot/src/main/java/org/mobilitydata/gtfsvalidator/springboot/GtfsValidatorController.java

```diff
@@ -0,0 +1,356 @@
+      @RequestParam(required = false, defaultValue = DEFAULT_COUNTRY_CODE) String country_code,
+      @RequestParam() String url,
+      @RequestParam(required = false, defaultValue = Arguments.VALIDATION_REPORT_NAME_JSON)
+          String validation_report_name,
+      @RequestParam(required = false, defaultValue = Arguments.SYSTEM_ERRORS_REPORT_NAME_JSON)
+          String system_error_report_name,
+      @RequestParam(required = false, defaultValue = "dataset id value") String dataset_id,
+      @RequestParam(required = false, defaultValue = "commit sha value") String commit_sha) {
+
+    Arguments args =
+        queryParametersToArguments(
+            output_base,
+            threads,
+            country_code,
+            url,
+            validation_report_name,
+            system_error_report_name);
+    StringBuilder messageBuilder = new StringBuilder();
+    HttpStatus status;
+    final long startNanos = System.nanoTime();
+
+    if (!CliParametersAnalyzer.isValid(args)) {
+      status = HttpStatus.BAD_REQUEST;
+      messageBuilder.append(
+          "%Bad request. Please check query parameters and execution logs for more information.\n");
+      return new ResponseEntity<>(buildResponseBody(messageBuilder.toString(), status), status);
+    }
+    GtfsInput gtfsInput;
+    GtfsFeedContainer feedContainer;
+    ValidatorLoader validatorLoader;
+    NoticeContainer noticeContainer = new NoticeContainer();
+    try {
+      validatorLoader = new ValidatorLoader();
+    } catch (ValidatorLoaderException e) {
+      return generateResponse(
+          messageBuilder.append(e.getMessage()).toString(), HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+    GtfsFeedLoader feedLoader = new GtfsFeedLoader();
+    try {
+      gtfsInput = createGtfsInput(args);
+    } catch (IOException ioException) {
+      logger.atSevere().withCause(ioException);
+      return generateResponseAndExportReport(
+          messageBuilder.append(ioException.getMessage()).toString(),
+          HttpStatus.BAD_REQUEST,
+          noticeContainer,
+          args);
+    } catch (URISyntaxException uriSyntaxException) {
+      logger.atSevere().withCause(uriSyntaxException);
+      messageBuilder.append("Internal error. Syntax error in URI.\n");
+      return generateResponseAndExportReport(
+          messageBuilder.append(uriSyntaxException.getMessage()).toString(),
+          HttpStatus.BAD_REQUEST,
+          noticeContainer,
+          args);
+    }
+    ValidationContext validationContext =
+        ValidationContext.builder()
+            .setCountryCode(
+                CountryCode.forStringOrUnknown(
+                    args.getCountryCode() == null ? CountryCode.ZZ : args.getCountryCode()))
+            .setCurrentDateTime(new CurrentDateTime(ZonedDateTime.now(ZoneId.systemDefault())))
+            .build();
+
+    try {
+      feedContainer =
+          Main.loadAndValidate(
```

Comment:
Thanks for flagging that, I implemented websocket via springboot, will take this into consideration when deploying to Kubernetes (will likely follow the steps described [here](https://medium.com/k8scaleio/running-websocket-app-on-kubernetes-2e13eabb4c4f)).

Reply:
>Thanks for flagging that, I implemented WebSocket via springboot (addressed in https://github.com/MobilityData/gtfs-validator/pull/933/commits/57f3e71953a771a131a460caa7c4f29e86ff7886) and will take this into consideration when deploying to Kubernetes (will likely follow the steps described [here](https://medium.com/k8scaleio/running-websocket-app-on-kubernetes-2e13eabb4c4f)).

I've never used Spring Boot myself, but I'm not sure this is enough to "implement" web sockets? For example see https://stackoverflow.com/questions/50984497/how-to-write-websockets-for-rest-calls-in-spring-boot.

This code in this PR looks like it's using annotations to handle incoming HTTP GET requests in the REST model and then provide a response. I imagine you'd need to re-write this using the classes in the Spring Boot `web.socket` package, as websockets don't fit in the request/response REST model.

As I said above, I don't know if websockets are necessary, but if you use the REST request/response model you'll have clients waiting for a long time for responses because it's synchronous code, so the validation needs to complete before the server can return the response. I'd test first and see if this is a problem if it involves a lot of effort to re-write this for websockets.